tuning/data/data_processors.py

```diff
@@ -238,19 +249,12 @@ def process_dataset_configs(
         return train_dataset
 
 
-def autoregister_available_handlers(processor: DataPreProcessor):
-    if processor is None:
-        return
-    for name, func in AVAILABLE_DATA_HANDLERS.items():
-        processor.register_data_handler(name=name, func=func)
-
-
 def get_datapreprocessor(
     processor_config: DataPreProcessorConfig, tokenizer: AutoTokenizer
 ) -> DataPreProcessor:
     processor = DataPreProcessor(
         processor_config=processor_config,
         tokenizer=tokenizer,
     )
-    autoregister_available_handlers(processor)
+    processor.register_data_handlers(AVAILABLE_DATA_HANDLERS)
```

Comment:
If this function is called twice, won't this be added twice?

Reply:
yes..added a warning if the code updates an existing handler

Reply:
yes..added a warning as you mentioned for this scenario

Reply:
I think for default handlers we should just add it upfront or have a flag to ensure it does not get added twice irrespective of how many times it is called.